### ISSUE TITLE
Arguments to CompositeDisposable.add must have a .dispose() method

### ISSUE THREAD

**Issue by antibagr** [codota/tabnine-atom#32]
i've been experiencing a problems using TabNine on Atom, although it's an awesome tools, it's seems I've got a bug.

It was thrown after I've added new rule for Windows Firewall to allow 55144 port for an outgoing requests. 

**Atom**: 1.51.0 x64
**Electron**: 5.0.13
**OS**: Microsoft Windows 10 Home Single Language
**Thrown From**: [TabNine](ssh://git@github.com/codota/tabnine-atom) package 2.9.0

### Stack Trace

Failed to activate the TabNine package

```
At Arguments to CompositeDisposable.add must have a .dispose() method

TypeError: Arguments to CompositeDisposable.add must have a .dispose() method
    at assertDisposable (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:1193843)
    at CompositeDisposable.add (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:1194811)
    at Object.consumeProvider (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:162524)
    at Object.consumeProvider_2 (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:162222)
    at Provider.provide (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:14:1086106)
    at ServiceHub.provide (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:3283140)
    at Package.activateServices (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:3291426)
    at Package.activateNow (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:3288402)
    at ~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:3287726
    at Package.measure (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:3285336)
    at ~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:3287586
    at new Promise (<anonymous>)
    at Package.activate (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:3287529)
    at PackageManager.activatePackage (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:383037)
    at ~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:378203
    at Array.forEach (<anonymous>)
    at ~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:378187
    at ~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:335143
    at Function.simpleDispatch (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:1190297)
    at Emitter.emit (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:1191738)
    at Config.emitChangeEvent (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:337631)
    at Config.setRawValue (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:334871)
    at Config.set (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:330665)
    at Config.removeAtKeyPath (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:332688)
    at Package.enable (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:3285121)
    at PackageManager.enablePackage (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:11:375806)
    at HTMLButtonElement.enablementButtonClickHandler (~/AppData/Local/atom/app-1.51.0/resources/app/static/<embedded>:14:2571661)
```

### Commands

```
     -6:08.3.0 snippets:previous-tab-stop (input.hidden-input)
     -6:08.3.0 editor:outdent-selected-rows (input.hidden-input)
     -6:07.9.0 core:save (input.hidden-input)
     -6:06.3.0 editor:toggle-line-comments (input.hidden-input)
     -6:05.4.0 core:save (input.hidden-input)
     -6:01.8.0 editor:toggle-line-comments (input.hidden-input)
     -6:01.5.0 core:save (input.hidden-input)
     -6:00.6.0 core:close (input.hidden-input)
     -5:58.2.0 code-format:format-code (input.hidden-input)
     -5:54.1.0 core:save (input.hidden-input)
     -3:36.6.0 core:select-all (input.hidden-input)
     -3:36.5.0 core:paste (input.hidden-input)
     -3:35.6.0 core:confirm (input.hidden-input)
     -3:35.1.0 core:backspace (input.hidden-input)
  3x -3:34.7.0 core:confirm (input.hidden-input)
     -3:25.6.0 core:backspace (input.hidden-input)
```

### Non-Core Packages

```
atom-ide-debugger-python 0.7.3 
atom-ide-ui 0.13.0 
atom-python-run 0.9.7 
autocomplete-python 1.16.0 
ide-python 1.5.0 
kite 0.191.0 
python-mrigor 2.1.3 
script 3.26.0 
split-diff 1.6.1 
TabNine 2.9.0 
teletype 0.13.4 
```

TypeError: Arguments to CompositeDisposable.add must have a .dispose() method
    at assertDisposable (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:1193843)
    at CompositeDisposable.add (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:1194811)
    at Object.consumeProvider (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:162524)
    at Object.consumeProvider_2 (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:162222)
    at Provider.provide (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:14:1086106)
    at ServiceHub.provide (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:3283140)
    at Package.activateServices (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:3291426)
    at Package.activateNow (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:3288402)
    at C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:3287726
    at Package.measure (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:3285336)
    at C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:3287586
    at new Promise (<anonymous>)
    at Package.activate (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:3287529)
    at PackageManager.activatePackage (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:383037)
    at C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:378203
    at Array.forEach (<anonymous>)
    at C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:378187
    at C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:335143
    at Function.simpleDispatch (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:1190297)
    at Emitter.emit (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:1191738)
    at Config.emitChangeEvent (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:337631)
    at Config.setRawValue (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:334871)
    at Config.set (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:330665)
    at Config.removeAtKeyPath (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:332688)
    at Package.enable (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:3285121)
    at PackageManager.enablePackage (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:11:375806)
    at HTMLButtonElement.enablementButtonClickHandler (C:\Users\Лариса\AppData\Local\atom\app-1.51.0\resources\app\static\<embedded>:14:2571661)
The error was thrown from the TabNine package. You can help by creating an issue. Please explain what actions triggered this error.

![tabnine-Window](https://user-images.githubusercontent.com/16188591/95601770-9c457a00-0a5c-11eb-8473-1f6d5dd8f13f.png)

**Comment by GoMapur** [comment]
+1

**Comment by dimacodota** [comment]
fixed in v4.0.0
